sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -22,8 +28,31 @@ impl BearerTokenCredentialPolicy {
         Self {
             credential,
             scopes: scopes.into_iter().map(|s| s.into()).collect(),
+            access_token: Arc::new(Mutex::new("UNSET".to_string())), // Need a better default here or use Option
+            last_refresh_time: Arc::new(Mutex::new(0)), // Need a better default here aswell
         }
     }
+
+    fn update_data(&self, new_access_token: &str) {
+        let mut my_access_token = self.access_token.lock().unwrap();
+        *my_access_token = new_access_token.to_string();
+        let mut my_last_refresh_time = self.last_refresh_time.lock().unwrap();
+        *my_last_refresh_time = OffsetDateTime::now_utc().unix_timestamp();
+    }
+
+    fn should_refresh(&self) -> bool {
+        let current_time = OffsetDateTime::now_utc().unix_timestamp();
+        let my_last_refresh_time = self.last_refresh_time.lock().unwrap();
```

Comment:
Never `unwrap()` in production code. This should have a better default e.g., `unwrap_or_default()` or the method needs to return a `std::result::Result<T, E>`.